### PR TITLE
Doc/def pass

### LINKED ISSUE #48: Add `ZZPhaseToRz` at optimization level 2
See https://github.com/CQCL/pytket-quantinuum/pull/45#issue-1394620829.

### PR DESCRIPTION
**Pull request by CalMacCQ**
The old `default_compilation_pass` info table was out of date. This could explain some confusion. See the updated table below.

Screenshot 2022-10-18 at 11 17 04

Is there a reason that `ZZPhaseToRz` is not performed at optimisation_level=2? (edit: this was fixed #48 )

Megan also sent me some updates on the batching feature which I'll add shortly in another commit.

Let me know if anything else on this page seems out of date. Note that the web portal link and the logo will up updated in a forthcoming release.

Cheers

### CODE REVIEW

File: docs/intro.txt [hunk 1]
```diff
@@ -59,27 +59,40 @@ The passes applied by different levels of optimisation are specified in the tabl
...
 
 * [1] If no value is specified then ``optimisation_level`` defaults to a value of 1.
 
 * [2] ``self.rebase_pass`` is a rebase that converts the circuit to the Quantinuum native gate set (e.g. {Rz, PhasedX, ZZMax, ZZPhase}).
 
-* [3] ``auto_squash_pass`` has arguments ``auto_squash_pass({OpType.PhasedX,OpType.Rz})``
+* [3] ``FullPeepholeOptimise`` has the argument ``target_2qb_gate=OpType.TK2``.
 
-* [4] ``SimplifyInitial`` has arguments  ``SimplifyInitial(allow_classical=False, create_all_qubits=True, xcirc=_xcirc)``. Here ``_xcirc`` is a single qubit circuit that implements the ``X`` gate using the native ``PhasedX`` gate.
+* [4] Gate fidelity information is passed into ``DecomposeTK2`` as ``DecomposeTK2(**fidelities)``. This allows for noise aware decompositions of TK2 gates.
```
Comment (ss2165) [inline docs/intro.txt:91]: we don't have gate fidelity information for quantinuum though right?

**Review by cqc-melf — COMMENTED** [review]
Happy to approve this now, any more comments from @cqc-alec @ss2165 ?

**Comment by cqc-alec** [comment]
> Happy to approve this now, any more comments from @cqc-alec @ss2165 ?

Fine with me.

File: docs/intro.txt [hunk 2]
```diff
@@ -68,27 +68,38 @@ The passes applied by different levels of optimisation are specified in the tabl
...
+     - `RemoveRedundancies <https://cqcl.github.io/tket/pytket/api/passes.html#pytket.passes.RemoveRedundancies>`_
+     - `RemoveRedundancies <https://cqcl.github.io/tket/pytket/api/passes.html#pytket.passes.RemoveRedundancies>`_
    * -
-     -
-     - `SimplifyInitial [4] <https://cqcl.github.io/tket/pytket/api/passes.html#pytket.passes.SimplifyInitial>`_
+     - `auto_squash_pass [5] <https://cqcl.github.io/tket/pytket/api/passes.html#pytket.passes.auto_rebase.auto_squash_pass>`_
+     - `auto_squash_pass [5] <https://cqcl.github.io/tket/pytket/api/passes.html#pytket.passes.auto_rebase.auto_squash_pass>`_
+   * -
+     - `SimplifyInitial [6] <https://cqcl.github.io/tket/pytket/api/passes.html#pytket.passes.SimplifyInitial>`_
+     - `SimplifyInitial [6] <https://cqcl.github.io/tket/pytket/api/passes.html#pytket.passes.SimplifyInitial>`_
```
Comment (ss2165) [inline docs/intro.txt:92]: The references are now incorrect because one of them has been removed so the numbers have shifted

**Review by ss2165 — CHANGES_REQUESTED** [review]

File: docs/intro.txt (hunk 2 above)
Comment (CalMacCQ) [reply docs/intro.txt]: Thanks, should be fixed now

**Review by ss2165 — APPROVED** [review]

**Pull request merged**